### PR TITLE
Add beginners tutorial

### PR DESCRIPTION
**Pull request by Nikolaiam**
Spacemacs has good documentation for the more computer-savvy, but
for non-programmers and those who have never heard of Emacs or Vim
before the process of installation and initial setup can be a bit
overwhelming, especially as a the information has to be collected from
multiple sources.

This detailed tutorial walks the user through the
setup process and shows the basic functionality of the program. It is
meant to serve as the one document one needs to get up-and-running with
Spacemacs.

Feedback is very welcome!

### CODE REVIEW

File: doc/BEGINNERS_TUTORIAL.org [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+called "frames". A frame is what pops up when you launch Spacemacs from your
+desktop shortcut. A frame contains windows and buffers.
+
+Windows in spacemacs are the visual spaces a frame is divided into. The default
+is one, but windows can be split to allow editing multiple files in one frame.
+Let's try this. Press SPC to bring up the menu. You can see different letters
+having different submenus associated with them, usually with a mnemonic for
+easier recall. The letter w is assigned for "windows": press it. A new menu
+opens with further options. We will write the character / (that is, shift and
+seven) to split the currently active window vertically into two.
```
Comment (cpaulik) [inline doc/BEGINNERS_TUTORIAL.org]: That is only true for certain keyboard layouts.

File: doc/BEGINNERS_TUTORIAL.org [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+the (hidden) .git folder in the same folder the tracked file(s) are in. You can
+make further commits the same way.
+** Daemon mode and instant startup (Linux) 
+Emacs can be used in daemon mode: a daemon runs in the background and launches
+clients. This way new frames launch instantly without delay.
+https://www.emacswiki.org/emacs/EmacsAsDaemon tells more about the daemon
+and how to set it to launch automatically on startup.
+** Swap caps lock and esc keys on your keyboard
+This is useful outside of Spacemacs as well!
+** Troubleshooting
```
Comment (cpaulik) [inline doc/BEGINNERS_TUTORIAL.org]: Maybe it would be best to link to the FAQ here.

File: doc/BEGINNERS_TUTORIAL.org [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+Now you should see two windows of this tutorial, and the one on the left should
+be active, as can be seen from the modeline in the bottom or by moving the
+cursor around using the navigation keys. This isn't very useful, as we
+would probably want to see a different file on the right.
+
+First, activate the window on the right with SPC 2. Now that the window on the 
+right is active, we can open a different buffer for a different file. We'll use
+the scratch buffer, which can be used like a notepad. Be warned, unlike other buffers it doesn't prompt you 
+whether you want to save the changes you've made when quitting Spacemacs!
+Press SPC b to open the buffers menu and then switch to the scratch buffer. Now
```
Comment (syl20bnr) [inline doc/BEGINNERS_TUTORIAL.org]: missing the `s` key to actually switch to the scratch buffer.

File: doc/BEGINNERS_TUTORIAL.org [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+keeping notes, maintaining todo lists, planning projects, and authoring
+documents with a fast and effective plain-text system", but this gives only a
+small inkling of its versatility. If you do any kind of writing at all, chances
+are that Org mode will make it easier and more fun. This tutorial was written in
+Org mode.
+
+Install the Org layer and open this tutorial. Press shift-TAB repeatedly and observe that
+this cycles the visibility of the contents of different headlines. Press t in
+normal mode and observe that you can add TODO tags on headlines. Press M-k
+(Alt-k, M stands for "modifier") or M-j in normal mode and see how you can
```
Comment (syl20bnr) [inline doc/BEGINNERS_TUTORIAL.org]: M stands for meta.

File: doc/BEGINNERS_TUTORIAL.org [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+Install the git layer, restart Spacemacs and open a file you want to version
+control. You can check the status of your file by pressing SPC g s. Select the
+folder your file is in. You will be prompted whether you want to create a
+repository in the folder. Select yes. You will see a list of "Untracked files":
+navigate to the file you want to track and press s to "stage changes". You might
+be prompted to save the file: save it if necessary. Now the new file needs to be
+commited: press c and c again. Two windows pop up: one showing the changes
+you've made since the last edit (in this case, the whole document) and another
+prompting for a commit message. Write "Initial commit", press ESC to exit back
+to normal mode and press :wq to write and quit the commit message.
```
Comment (syl20bnr) [inline doc/BEGINNERS_TUTORIAL.org]: instead of `:wq` most people use `SPC m c` to confirm and `SPC m a` to abort or `C-c C-c` and `C-c C-k`.

**Comment by bmag** [comment]
I like it. A few notes:
- key shortcuts look nicer if you format them. e.g. `~SPC b b~` instead of `SPC b b`
- absolute beginners don't know how to read Emacs key notation: `SPC` is space, `C-` is Ctrl, `M-` is Alt (or something else on Mac?), `s-` is Super, etc. `C-c a` means "Ctrl+c, then a", `C-c C-a` means "Ctrl+c, then Ctrl+a", etc. I couldn't find a good external resource that explains it, although I'm sure I've read one in the past.

File: doc/BEGINNERS_TUTORIAL.org [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+ and install the Spacemacs extension files, which is easiest done by using a program
+ called Git. The steps are easy and outlined below.
+** 1. Install Emacs
+Follow the instructions in https://github.com/syl20bnr/spacemacs#prerequisites.
+** 2. Install Git
+Go to https://git-scm.com/downloads and install Git on your platform.
+** 3. Install Spacemacs
+Open a terminal or command prompt, paste the following code to it:
+
+git clone https://github.com/syl20bnr/spacemacs ~/.emacs.d
```
Comment (d12frosted) [inline doc/BEGINNERS_TUTORIAL.org]: Should be wrapped in `BEGIN_SRC` as:

```
#+BEGIN_SRC shell
$ git clone https://github.com/syl20bnr/spacemacs ~/.emacs.d
#+END_SRC

File: doc/BEGINNERS_TUTORIAL.org [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+  scientific papers... 
+- Powerful modes for programming in dozens of programming languages
+- Deeply customizable yet beginner-friendly
+* Installation
+ Spacemacs is a beginner-friendly and powerful distribution of a popular text
+ editor called Emacs. To install Spacemacs you need to first install base Emacs and then download
+ and install the Spacemacs extension files, which is easiest done by using a program
+ called Git. The steps are easy and outlined below.
+** 1. Install Emacs
+Follow the instructions in https://github.com/syl20bnr/spacemacs#prerequisites.
```
Comment (bmag) [inline doc/BEGINNERS_TUTORIAL.org]: I think we should point out that Windows users need Gnu-TLS support for HTTPS to work (although it's possible to use HTTP instead). The recommended 64-bit build has it (http://emacsbinw64.sourceforge.net/), but the official 32-bit build doesn't (https://www.gnu.org/software/emacs/download.html).
If necessary, Gnu-TLS can be installed from http://www.gnutls.org/download.html. To check if Emacs supports Gnu-TLS:
```
M-: (gnutls-available-p)
```

File: doc/BEGINNERS_TUTORIAL.org [hunk 8]
```diff
@@ -0,0 +1,184 @@
...
+** 2. Install Git
+Go to https://git-scm.com/downloads and install Git on your platform.
+** 3. Install Spacemacs
+Open a terminal or command prompt, paste the following code to it:
+
+git clone https://github.com/syl20bnr/spacemacs ~/.emacs.d
+
+Press enter to execute the code and the program you installed in the previous step,
+Git, will download the Spacemacs extension files.
+** 4. Install the default font
```
Comment (d12frosted) [inline doc/BEGINNERS_TUTORIAL.org]: Very good point. 👍 💯

File: doc/BEGINNERS_TUTORIAL.org (hunk 6 above)
Comment (bmag) [inline doc/BEGINNERS_TUTORIAL.org]: I don't think `~/.emacs.d` works in Windows (unless you use cygwin or similar). Easiest way I found to discover the real `.emacs.d` path on windows is to open `~` in Dired: `M-x dired ~/`

File: doc/BEGINNERS_TUTORIAL.org [hunk 9]
```diff
@@ -0,0 +1,184 @@
...
+The default font used by Spacemacs is Source Code Pro by Adobe. It is
+recommended to [[https://github.com/adobe-fonts/source-code-pro][install]] it on your system to ensure correct visual representation.
+** Done!
+Open Spacemacs by clicking the Emacs icon in your applications menu. The first
+time Spacemacs launches, it will ask you for your preferred editing style. We
+recommend selecting the default option, "Evil", more of which is talked about in
```
Comment (d12frosted) [inline doc/BEGINNERS_TUTORIAL.org]: I would recommend either `evil` or `emacs` based on user background. Emacs users trying Spacemacs for the first time without any vim background should not use `evil` unless they are here for `evil` part. 

I think the main point I would like to make here is that user might choose any editing style based on his preference and background. `hybrid` style is generally for advanced users. **But** it's very important to say that any decisions made here can be easily changed later on.

File: doc/BEGINNERS_TUTORIAL.org [hunk 10]
```diff
@@ -0,0 +1,184 @@
...
+the next section. For troubleshooting, see the last section.
+* Getting started
+** Modal text editing - why and how?
```
Comment (bmag) [inline doc/BEGINNERS_TUTORIAL.org]: Not usre, but maybe we should note that we also support a non-modal editing style - `emacs` (with regular Emacs key bindings)

File: doc/BEGINNERS_TUTORIAL.org (hunk 6 above)
Comment (d12frosted) [inline doc/BEGINNERS_TUTORIAL.org]: Ah, I don't know how it works on Windows (never really used it oO).

My point was about document structure :)

**Comment by d12frosted** [comment]
General advice - don't forget to wrap inline code and key bindings with tilde like `~code~`. Also you should wrap code blocks with following:

```
#+BEGIN_SRC lang-name

#+END_SRC
```

It renders beautifully in `org-mode` and on GitHub Pages.

File: doc/BEGINNERS_TUTORIAL.org (hunk 10 above)
Comment (d12frosted) [inline doc/BEGINNERS_TUTORIAL.org]: YES, YES and YES! Not everyone use Spacemacs for `evil` part. Remove `emacs` editing style and you'll make it impossible for me to use Spacemacs.

**Comment by d12frosted** [comment]
Also, please read and follow [these conventions](https://github.com/syl20bnr/spacemacs/blob/develop/doc/CONVENTIONS.org#writing-documentation) about writing documentation. 😸

File: doc/BEGINNERS_TUTORIAL.org [hunk 11]
```diff
@@ -0,0 +1,184 @@
...
+The modal editing features of Spacemacs originate from a text editor called VI,
+and thus the modal editing tutorial is called eVIl tutor. Press SPC h T (that
+is, the spacebar followed by h and T) to familiarize yourself with 
+modal editig. 
+** Using the spacebar to launch commands
+Now that you are familiar with writing and editing text it is time to put the "Space"
+into Spacemacs. Because the spacebar is the most accessible key on the keyboard
+and is pressed by the strongest fingers (the thumbs), it is a natural choice for
+launching commands. You can think of it as the start menu of Spacemacs.
+** Buffers, windows and frames
```
Comment (d12frosted) [inline doc/BEGINNERS_TUTORIAL.org]: Very nice to bring up this topic. 💯

File: doc/BEGINNERS_TUTORIAL.org [hunk 12]
```diff
@@ -0,0 +1,184 @@
...
+Open a terminal or command prompt, paste the following code to it:
+
+git clone https://github.com/syl20bnr/spacemacs ~/.emacs.d
+
+Press enter to execute the code and the program you installed in the previous step,
+Git, will download the Spacemacs extension files.
+** 4. Install the default font
+The default font used by Spacemacs is Source Code Pro by Adobe. It is
+recommended to [[https://github.com/adobe-fonts/source-code-pro][install]] it on your system to ensure correct visual representation.
+** Done!
```
Comment (bmag) [inline doc/BEGINNERS_TUTORIAL.org]: Almost done. Part of the installation process (IMO) is to run Spacemacs for the first time, choose editing style and distribution (and helm/ivy?), then wait for Spacemacs to download lots of packages. After that, restart Spacemacs (to get a clean state) and only then the installation is complete.

File: doc/BEGINNERS_TUTORIAL.org [hunk 13]
```diff
@@ -0,0 +1,184 @@
...
+and is pressed by the strongest fingers (the thumbs), it is a natural choice for
+launching commands. You can think of it as the start menu of Spacemacs.
+** Buffers, windows and frames
+Because of its roots in Emacs, which was developed in the 80's
+before the advent of modern graphical user interfaces, Spacemacs has 
+a different name of what we normally call "windows": in Spacemacs these are
+called "frames". A frame is what pops up when you launch Spacemacs from your
+desktop shortcut. A frame contains windows and buffers.
+
+Windows in spacemacs are the visual spaces a frame is divided into. The default
```
Comment (d12frosted) [inline doc/BEGINNERS_TUTORIAL.org]: > Windows in spacemacs

First of all, 'Spacemacs' and not 'spacemacs' 😸 

Also, I think it's better to use 'Emacs' instead of 'Spacemacs' when you are talking about general behaviour or concepts that are not really related to Spacemacs. The main point is to avoid confusion about what Spacemacs is.

File: doc/BEGINNERS_TUTORIAL.org [hunk 14]
```diff
@@ -0,0 +1,184 @@
+* Why Spacemacs?
+- Unparallelled text and structure editing for all types of writing tasks: creative writing, blogging, note-taking, todo-lists,
+  scientific papers... 
+- Powerful modes for programming in dozens of programming languages
+- Deeply customizable yet beginner-friendly
+* Installation
+ Spacemacs is a beginner-friendly and powerful distribution of a popular text
```
Comment (d12frosted) [inline doc/BEGINNERS_TUTORIAL.org]: > Spacemacs is ... distribution

I wonder if we should call it 'distribution'. Because people often confuse Spacemacs with replacement for Emacs, while it's actually extension, a set of configurations. But I am not native English speaker, so that might be my problem in understanding what 'distribution' actually means.

Comment (Nikolaiam) [inline doc/BEGINNERS_TUTORIAL.org]: I think you're right, extension is a better description of what Spacemacs is!

File: doc/BEGINNERS_TUTORIAL.org (hunk 7 above)
Comment (Nikolaiam) [inline doc/BEGINNERS_TUTORIAL.org]: I'm somewhat unfamiliar with this topic, but do you think this information is best placed in the beginners tutorial? If I understood correctly, the recommended build has this and http can be used, so by following these instructions, no problems should arise? I'm just thinking that the more concise the instructions are, the better, and maybe this information would find a better place [https://github.com/syl20bnr/spacemacs#prerequisites](here)?

**Comment by Nikolaiam** [comment]
After the newest amendments the Travis CI build failed and I'm at a total loss why. Any suggestions on what the problem is?

**Comment by syl20bnr** [comment]
This is my fault, will fix this evening.

**Comment by syl20bnr** [comment]
Thank you ! 👍 
Cherry-picked into develop branch, you can safely delete your branch.
